src/devtools/views/right-rail/index.js

```diff
@@ -17,16 +24,49 @@ class RightRail extends Presenter {
     }
   }
 
+  initDrag = e => {
+    startX = e.clientX
+    startWidth = this.state.paneWidth
+
+    document.documentElement.addEventListener('mousemove', this.onDrag, false)
+    document.documentElement.addEventListener('mouseup', this.stopDrag, false)
+  }
+
+  onDrag = e => {
+    this.setState({
+      paneWidth: startWidth - (e.clientX - startX)
+    })
+  }
+
+  stopDrag = e => {
+    document.documentElement.removeEventListener(
+      'mousemove',
+      this.onDrag,
+      false
+    )
+    document.documentElement.removeEventListener(
+      'mouseup',
+      this.stopDrag,
+      false
+    )
+  }
+
   render() {
     const { open } = this.props
     const { snapshot, focused } = this.model
+    const { paneWidth } = this.state
 
     if (!open) {
       return null
     }
 
     return (
-      <aside className={css.container}>
+      <aside
+        className={css.container}
+        style={{ 'min-width': paneWidth + 'px' }}
+      >
+        <div className={css.resizer} onMouseDown={this.initDrag} />
+
         <section>
           <header className={css.header}>
             <h2 className={css.title}>State</h2>
```

Comment:
Assuming the rails fill their container, could this be a general purpose component that wraps both rails? Then you could share the code between each:

```
<Resizeable defaultWidth={0.25}>
  <LeftRail />
</Resizeable>
```